compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -547,8 +547,22 @@ object TreeChecker {
         i"owner chain = ${tree.symbol.ownersIterator.toList}%, %, ctxOwners = ${ctx.outersIterator.map(_.owner).toList}%, %")
     }
 
+    private def checkParents(tree: untpd.TypeDef)(using Context): Unit = {
+      val TypeDef(_, impl: Template) = tree: @unchecked
+      assert(ctx.owner.isInstanceOf[ClassSymbol])
+      val symbolParents = ctx.owner.asClass.info.parents
+      val treeParents = impl.parents.map(_.tpe)
+      assert(symbolParents.toSet == treeParents.toSet,
+      i"""TypeDef tree have different parents than its Symbol
+         |Symbol ${ctx.owner} has the following parents $symbolParents registred.
+         |The tree below has the following parents $treeParents registred:
+         |$tree
+      """.stripMargin)
+    }
+
     override def typedTypeDef(tdef: untpd.TypeDef, sym: Symbol)(using Context): Tree = {
       assert(sym.info.isInstanceOf[ClassInfo | TypeBounds], i"wrong type, expect a template or type bounds for ${sym.fullName}, but found: ${sym.info}")
+      checkParents(tdef)
```

Comment:
Do we need this one? Don't we call `typedClassDef` in `super.typedTypeDef(tdef, sym)` bellow?

Reply:
No, we don't